Bottom: 1b0eda45b6430ad1ba50d03aa1beddc8a628aa65
Top:    b1ad32563972d70c7a93d5dfe1e6de9eee4512e0
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-04 14:32:02 -0700

Refresh of rs-maxseg

---

diff --git a/src/rsocket.c b/src/rsocket.c
index f5ebae4..90502d3 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1818,12 +1818,10 @@ int rgetsockopt(int socket, int level, int optname,
 			*optlen = sizeof(int);
 			break;
 		case TCP_MAXSEG:
-			if (rs->cm_id && rs->cm_id->route &&
-			    rs->cm_id->route.num_paths) {
-				*((int *) optval) = 1 << (7 + rs->cm_id->route.path_rec->mtu);
-			} else {
-				*((int *) optval) = 2048;
-			}
+			*((int *) optval) = (rs->cm_id && rs->cm_id->route &&
+					     rs->cm_id->route.num_paths) ?
+					    1 << (7 + rs->cm_id->route.path_rec->mtu) :
+					    2048;
 			*optlen = sizeof(int);
 			break;
 		default:
